log4j-core/src/main/java/org/apache/logging/log4j/core/impl/ThrowableProxy.java

```diff
@@ -401,15 +424,17 @@ public String getExtendedStackTraceAsString(
      * @param textRenderer The message renderer.
      * @param suffix Append this to the end of each stack frame.
      * @param lineSeparator The end-of-line separator.
+     * @param linesToKeep The total line count of final result
      */
     public void formatExtendedStackTraceTo(
             final StringBuilder sb,
             final List<String> ignorePackages,
             final TextRenderer textRenderer,
             final String suffix,
-            final String lineSeparator) {
+            final String lineSeparator,
+            final Integer linesToKeep) {
```

Comment:
Same as above: you can add a new method, but the old method must remain.

Reply:
updated